pandas/_libs/src/datetime/np_datetime_strings.c

```diff
@@ -138,59 +89,6 @@ int parse_iso_8601_datetime(char *str, int len,
     out->month = 1;
     out->day = 1;
 
-    /*
-     * The string "today" means take today's date in local time, and
-     * convert it to a date representation. This date representation, if
-     * forced into a time unit, will be at midnight UTC.
-     * This is perhaps a little weird, but done so that the
-     * 'datetime64[D]' type produces the date you expect, rather than
-     * switching to an adjacent day depending on the current time and your
-     * timezone.
-     */
-    if (len == 5 && tolower(str[0]) == 't' && tolower(str[1]) == 'o' &&
-        tolower(str[2]) == 'd' && tolower(str[3]) == 'a' &&
-        tolower(str[4]) == 'y') {
-        NPY_TIME_T rawtime = 0;
-        struct tm tm_;
-
-        time(&rawtime);
-        if (get_localtime(&rawtime, &tm_) < 0) {
-            return -1;
-        }
-        out->year = tm_.tm_year + 1900;
-        out->month = tm_.tm_mon + 1;
-        out->day = tm_.tm_mday;
-
-        /*
-         * Indicate that this was a special value, and
-         * is a date (unit 'D').
-         */
-        if (out_local != NULL) {
-            *out_local = 0;
-        }
-
-        return 0;
-    }
-
-    /* The string "now" resolves to the current UTC time */
-    if (len == 3 && tolower(str[0]) == 'n' && tolower(str[1]) == 'o' &&
-        tolower(str[2]) == 'w') {
-        NPY_TIME_T rawtime = 0;
-
-        time(&rawtime);
-
-        /*
-         * Indicate that this was a special value, and
-         * use 's' because the time() function has resolution
-         * seconds.
-         */
-        if (out_local != NULL) {
-            *out_local = 0;
-        }
-
-        return convert_datetime_to_datetimestruct(PANDAS_FR_s, rawtime, out);
```

Comment:
Side-benefit: this is the only usage of convert_datetime_to_datetimestruct outside of np_datetime.c, so after this we can get rid of the extra layer there.